### PR TITLE
Update module digital_ocean_sshkey

### PR DESCRIPTION
**Pull request by pmarques**
##### ISSUE TYPE

- New Module Pull Request

##### COMPONENT NAME

- cloud/digital_ocean/digital_ocean_sshkey

##### SUMMARY

Add module to manage (create, delete and rename)  Digital Ocean SSH Keys

### CODE REVIEW

**Comment by pmarques** [comment]
Migrated from https://github.com/ansible/ansible-modules-core/pull/5275

**Comment by harneksidhu** [comment]
@pmarques Are you bringing this PR back up? I have just migrated my PR https://github.com/ansible/ansible/pull/19182 which is dependent on this.

**Comment by pmarques** [comment]
@harneksidhu This as not merged, right? I've migrated it but somehow it was closed :/

**Comment by harneksidhu** [comment]
@pmarques 

I closed my PR as I am unable to resolve the build errors. https://github.com/ansible/ansible/pull/19182

**Comment by BondAnthony** [comment]
@pmarques any plans to rebase this PR?

**Comment by pmarques** [comment]
@BondAnthony sorry for the delay

**Review by BondAnthony — APPROVED** [review]

File: lib/ansible/modules/cloud/digital_ocean/do_sshkeys.py
Comment (BondAnthony) [inline lib/ansible/modules/cloud/digital_ocean/do_sshkeys.py]: Should this be "identifier" instead of identified?

File: lib/ansible/modules/cloud/digital_ocean/do_sshkeys_facts.py
Comment (BondAnthony) [inline lib/ansible/modules/cloud/digital_ocean/do_sshkeys_facts.py]: Floating-ips or list all the keys in your account?

File: lib/ansible/modules/cloud/digital_ocean/do_sshkeys_facts.py
Comment (BondAnthony) [inline lib/ansible/modules/cloud/digital_ocean/do_sshkeys_facts.py]: Did you mean to link to the floating-ips documentation or should this be "https://developers.digitalocean.com/documentation/v2/#list-all-keys"

File: lib/ansible/modules/cloud/digital_ocean/do_sshkeys_facts.py
Comment (BondAnthony) [inline lib/ansible/modules/cloud/digital_ocean/do_sshkeys_facts.py]: Can import os be removed? I don't see any reference to it.

**Review by BondAnthony — CHANGES_REQUESTED** [review]
Just a few minor changes..

**Review by BondAnthony — APPROVED** [review]
Looks good and it works well

**Comment by ryansb** [comment]
Instead of a new module, is there a reason you can't update [digital_ocean_sshkey](https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/cloud/digital_ocean/digital_ocean_sshkey.py) to use the V2 API?

**Comment by pmarques** [comment]
@ryansb The idea came from this issue ansible/ansible-modules-core/issues/4219

**Comment by BondAnthony** [comment]
@ryansb what do you do when maintainers are no longer active in the community? Do you elect new individuals to take over coding and testing? I would like to see some of these modules shipped. I would love to help make this possible.

**Comment by pmarques** [comment]
Moved facts module to new PR https://github.com/ansible/ansible/pull/26365

**Comment by pmarques** [comment]
rebased to fix conflict...

File: lib/ansible/modules/cloud/digital_ocean/_digital_ocean_sshkey.py
Comment (BondAnthony) [inline lib/ansible/modules/cloud/digital_ocean/_digital_ocean_sshkey.py]: You need to change preview to "deprecated" to fix the failing test.

[Ansible metadata-block](https://docs.ansible.com/ansible/dev_guide/developing_modules_documenting.html#ansible-metadata-block)

**Comment by BondAnthony** [comment]
needs_revision

File: lib/ansible/modules/cloud/digital_ocean/_digital_ocean_sshkey.py
Comment (pmarques) [reply lib/ansible/modules/cloud/digital_ocean/_digital_ocean_sshkey.py]: ups, should be fixed!

Comment (BondAnthony) [reply lib/ansible/modules/cloud/digital_ocean/_digital_ocean_sshkey.py]: I believe you changed the wrong file.

Comment (pmarques) [reply lib/ansible/modules/cloud/digital_ocean/_digital_ocean_sshkey.py]: oh gosh, no comment...

**Comment by BondAnthony** [comment]
shipit

**Comment by BondAnthony** [comment]
@resmo Could you review this since it's a new module and requires two approvals.

File: lib/ansible/modules/cloud/digital_ocean/do_sshkeys.py
Comment (BondAnthony) [inline lib/ansible/modules/cloud/digital_ocean/do_sshkeys.py]: After our discussion on PR #18922 I think you should add support for oauth_token in this module.

**Review by BondAnthony — CHANGES_REQUESTED** [review]
Just trying to start a standard, understand that on a later date we will have the option to remove the old environment variables.

File: lib/ansible/modules/cloud/digital_ocean/_digital_ocean_sshkey.py
Comment (resmo) [inline lib/ansible/modules/cloud/digital_ocean/_digital_ocean_sshkey.py]: deprecated in 2.4

File: lib/ansible/modules/cloud/digital_ocean/do_sshkeys.py
Comment (resmo) [inline lib/ansible/modules/cloud/digital_ocean/do_sshkeys.py]: this syntax is not python 2.6 compatible. please add a positional index  'Bearer {0}'.format(api_token),

File: lib/ansible/modules/cloud/digital_ocean/do_sshkeys.py
Comment (resmo) [inline lib/ansible/modules/cloud/digital_ocean/do_sshkeys.py]: this syntax is not python 2.6 compatible. please add a positional index

File: lib/ansible/modules/cloud/digital_ocean/do_sshkeys.py
Comment (resmo) [inline lib/ansible/modules/cloud/digital_ocean/do_sshkeys.py]: this syntax is not python 2.6 compatible. please add a positional index

File: lib/ansible/modules/cloud/digital_ocean/do_sshkeys.py
Comment (resmo) [inline lib/ansible/modules/cloud/digital_ocean/do_sshkeys.py]: this syntax is not python 2.6 compatible. please add a positional index

File: lib/ansible/modules/cloud/digital_ocean/do_sshkeys.py
Comment (resmo) [inline lib/ansible/modules/cloud/digital_ocean/do_sshkeys.py]: this syntax is not python 2.6 compatible. please add a positional index

File: lib/ansible/modules/cloud/digital_ocean/do_sshkeys.py
Comment (resmo) [inline lib/ansible/modules/cloud/digital_ocean/do_sshkeys.py]: this syntax is not python 2.6 compatible. please add a positional index

File: lib/ansible/modules/cloud/digital_ocean/do_sshkeys.py
Comment (resmo) [inline lib/ansible/modules/cloud/digital_ocean/do_sshkeys.py]: this syntax is not python 2.6 compatible. please add a positional index

**Review by resmo — CHANGES_REQUESTED** [review]
Probably 2 things to consider are to add the `timeout` param o pass to `fetch_url()` and if you add a `validate_certs` bool param, it will also be used in fetch_url https://github.com/ansible/ansible/blob/devel/lib/ansible/module_utils/urls.py#L1018

**Comment by pmarques** [comment]
@resmo about add *timeout* and *validate_certs* to fetch_url I don't use them but I understand their usefulness. But them will take me a few more days to implement and test

**Comment by resmo** [comment]
@pmarques sure, take your time. ping me when ready for review :)

**Comment by pmarques** [comment]
@resmo please have a look, was more simple than I was expecting.

**Review by resmo — APPROVED** [review]

**Comment by resmo** [comment]
code LGTM

shipit

**Review by BondAnthony — APPROVED** [review]
It’s going to be nice to have a working ssh key module. Thanks!

shipit

**Comment by resmo** [comment]
bot_status

File: lib/ansible/modules/cloud/digital_ocean/do_sshkeys.py
Comment (ryansb) [inline lib/ansible/modules/cloud/digital_ocean/do_sshkeys.py]: The rename of this module isn't great - it breaks the convention of prefixing with `digital_ocean_`, would it be possible to name this `digital_ocean_ssh_keys` or `digital_ocean_sshkeys` so it matches the rest?

**Review by ryansb — CHANGES_REQUESTED** [review]

File: lib/ansible/modules/cloud/digital_ocean/do_sshkeys.py
Comment (pmarques) [reply lib/ansible/modules/cloud/digital_ocean/do_sshkeys.py]: @ryansb IMHO all that names are to be too big but I understand the problem of keep consistency. Can I suggest an alias to **digital_ocean_sshkeys**? (or rename to *digital_ocean_sshkeys* and create and alias to *do_sshkeys*)

Comment (pmarques) [reply lib/ansible/modules/cloud/digital_ocean/do_sshkeys.py]: btw, *do_* keeps the same convention from other modules like *cloudstack* = *cs_* ; "openstack" = *os_*

Comment (ryansb) [reply lib/ansible/modules/cloud/digital_ocean/do_sshkeys.py]: I'd much rather err on the side of longer names keeping convention, so naming the module `digital_ocean_sshkeys`.

Comment (pmarques) [reply lib/ansible/modules/cloud/digital_ocean/do_sshkeys.py]: @ryansb keep in mind that the API is slightly different.

Changes done:
* Rename to ** digital_ocean_sshkeys**
* Remove deprecation of old module (since we kept the name this doesn't make sense)

Comment (BondAnthony) [reply lib/ansible/modules/cloud/digital_ocean/do_sshkeys.py]: I don't believe the current digital_ocean_sshkey module works today since DO shutdown the v1 route.

**Comment by pmarques** [comment]
btw, this modules still tagged with **affects_2.3** should it be updated to **affects_2.4**

**Comment by Akasurde** [comment]
@pmarques Small nitpick - Could you please rename module to match 'digital_ocean_*' schema?

**Comment by pmarques** [comment]
@Akasurde what do you mean? is not that my last 2 commits?

**Review by pmarques — COMMENTED** [review]
@ryansb please, check it

**Review by Akasurde — APPROVED** [review]
LGTM

**Comment by BondAnthony** [comment]
@pmarques could you rebase please?

**Comment by pmarques** [comment]
@BondAnthony rebased, also added another commit to reflect the changes from 2425143cf22399e86ff6ecf3853c3162507fc496 by @abadger

**Comment by BondAnthony** [comment]
shipit

**Review by ryansb — APPROVED** [review]

**Comment by BondAnthony** [comment]
Thanks @ryansb

**Pull request merged**
